ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1182,13 +1182,23 @@ func (r *RayClusterReconciler) calculateStatus(ctx context.Context, instance *ra
 	// TODO (kevin85421): ObservedGeneration should be used to determine whether to update this CR or not.
 	newInstance.Status.ObservedGeneration = newInstance.ObjectMeta.Generation
 
-	runtimePods := corev1.PodList{}
-	filterLabels := client.MatchingLabels{utils.RayClusterLabelKey: newInstance.Name}
-	if err := r.List(ctx, &runtimePods, client.InNamespace(newInstance.Namespace), filterLabels); err != nil {
+	headPods := corev1.PodList{}
+	if err := r.List(ctx, &headPods, common.RayClusterHeadPodsAssociationOptions(instance).ToListOptions()...); err != nil {
```

Comment:
Can we reuse `isHeadPodRunningAndReady`?

https://github.com/ray-project/kuberay/blob/c099de4d7916f3232fb3f302595cad88fcc36253/ray-operator/controllers/ray/rayservice_controller.go#L1221

Reply:
By the way, RayService's status includes RayClusterStatus. If we provide information about whether the head Pod is ready, we won't need to retrieve the head Pod in the RayService controller. We would just need to read the CR status.

Reply:
One thing we need to keep in mind is that we must ensure the RayCluster always updates its status when the head's status changes, regardless of any errors thrown during reconciliation.